sycl/source/device.cpp

```diff
@@ -49,9 +50,16 @@ device::device(const device_selector &deviceSelector) {
 
 vector_class<device> device::get_devices(info::device_type deviceType) {
   vector_class<device> devices;
-  // Host device availability should not depend on the forced type
-  const bool includeHost =
-      detail::match_types(deviceType, info::device_type::host);
+  detail::device_filter_list *FilterList =
+      detail::SYCLConfig<detail::SYCL_DEVICE_FILTER>::get();
+  // Host device availability should depend on the forced type
+  bool includeHost;
+  // If SYCL_DEVICE_FILTER is set, we don't automatically include it.
+  // We will check if host devices are specified in the filter below.
+  if (FilterList)
+    includeHost = false;
```

Comment:
Shouldn't `FilterList->containsHost()` be used here to satisfy statement on line 58?

Reply:
I think we can use FilterList->containsHost() here instead of setting false.
Actually, line 76 will filter out HOST backend if Filter is not not backendCompatible() with HOST.
So, I will change as you suggested.